capstone/src/main/java/com/google/univiz/api/VisResource.java

```diff
@@ -1,7 +1,19 @@
 package com.google.univiz.api;
 
+import com.google.univiz.api.dataviz.Deadline;
+import com.google.univiz.api.dataviz.Timeline;
+import com.google.univiz.api.dataviz.TuitionStats;
+import java.util.List;
+
 /**
  * The VisResource interface is responsible for all data management/processing specific for the vis
  * feature of the UniViz webapp.
  */
-public interface VisResource {}
+public interface VisResource {
+
+  List<TuitionStats> getTuitionStats(List<String> colleges);
+
+  List<Timeline> getRecommendedTimeline(List<String> colleges);
+
+  List<Deadline> getDeadlines(List<String> colleges);
```

Comment:
wouldnt these all be taking in a ```List<CollegeData>```? i think it would, because that is the type that is going to have the information you want in the type that you want it.

Reply:
Ah ok I will change the methods to take in a CollegeData. Though I thought there was going to be something more [lightweight](https://github.com/googleinterns/univiz/pull/52#discussion_r465224050) seeing Michaela's suggestion and the [endorsements](https://github.com/googleinterns/univiz/pull/52#discussion_r465400360) of Chris and Kaveh. I had the impression that these methods would take in the list of colleges and then use CollegeData to fetch the JSON and then populate the appropriate data structures. But I guess it works if I directly take in a CollegeData object too.

Reply:
`CollegeData` isn't going to be involved in any parsing, i dont believe. but we can talk about this offline.